Tests is done

agoevnauruz opened the pull request.

agoevnauruz committed:
add dependency for allure report, selenide and junit

agoevnauruz committed:
final result

pavel-ulan commented on src/main/java/org/example/utils/YandexDriver.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Override
+    public WebDriver createDriver(Capabilities capabilities) {
+        System.setProperty("chromedriver", "/Users/mac/Downloads/chromedriver_mac_arm64");
```
**можно лучше**
для файлов типа драйверов для браузера стоит использовать папку property в проекте. тогда тот кто будет работать с твоим репозиторием сможет с ним работать.

pavel-ulan commented on src/test/java/RegistrationPageTest.java:
```diff
@@ -0,0 +1,94 @@
+                .isConfirmThatRegistrationFalls();
+    }
+
+    public String waitForLocalStorage() throws InterruptedException {
```
**надо исправить**
а этот метод тебе для чего? очень странное кастомное ожидание и если честно я так до конца и не понял для чего ты его используешь

agoevnauruz replied:
Метод waitForLocalStorage() используется для извлечения значения accessToken после успешной регистрации. Это значение затем передается в метод tearDown(), который удаляет созданного пользователя через userClient.delete(accessToken)

pavel-ulan commented on src/test/java/CheckMenuTest.java:
```diff
@@ -0,0 +1,66 @@
+                .clickLinkOrderFilling()
+                .isConfirmThatHeaderFillingIsDisplayed();
+
+        assertTrue("Transition to the menu section 'Filling' failed", isHeaderFillingDisplayed);
```
**надо исправить**
Логика в принципе понятна, но проблема в том, что эти элементы всегда displayed и поэтому твои тесты всегда ложноположительны (можешь проверить это если в дебаге перед проверкой проскролишь в другой раздел).
Изучи, что меняется в верстке при переходе по разделам, это должно помочь =)

agoevnauruz replied:
1. Проверять, что после перехода по ссылке в меню в верстке появляется ожидаемый заголовок раздела.

2. Проверять, что URL-адрес страницы изменился после перехода по ссылке в меню.

3. Проверять, что появились ожидаемые элементы верстки после перехода по ссылке в меню.

pavel-ulan reviewed (changes requested):
Тоже неплохо (использование Chain of invocation похвально, хотя и местами не очень элегантно использовано), но есть пару мест где ты применил очень странные решения. Посмотри комменты и поправь.

agoevnauruz committed:
Change test after review

agoevnauruz committed:
add some changes after review

Проверять, что после перехода по ссылке в меню в верстке появляется ожидаемый заголовок раздела.

Проверять, что URL-адрес страницы изменился после перехода по ссылке в меню.

Проверять, что появились ожидаемые элементы верстки после перехода по ссылке в меню.

agoevnauruz committed:
Delete CheckMenuTest.java

agoevnauruz committed:
change assert 

Check that after navigating to the link in the menu, the expected section header appears in the layout.

Check that the page URL has changed after navigating to the link in the menu.

Check that the expected layout elements are present after navigating to the link in the menu.

agoevnauruz committed:
add allure-report

agoevnauruz committed:
add changes after review

pavel-ulan reviewed (approved):
Да, сейчас валидно
